src/shared/hooks/reflect.ts

```diff
@@ -0,0 +1,131 @@
+import { useEffect, useState } from "react"
+import {
+  selectDisplayedRealmId,
+  selectStakingRealmId,
+  selectWalletName,
+  useDappSelector,
+} from "redux-state"
+import { usePresence, useSubscribe } from "@rocicorp/reflect/react"
+import { getClientState, reflectInstance } from "shared/utils"
+import { getRealmMapData } from "shared/constants"
+
+export function useReflect() {
+  const name = useDappSelector(selectWalletName)
+  const stakingRealmId = useDappSelector(selectStakingRealmId)
+  const realmMapData = stakingRealmId ? getRealmMapData(stakingRealmId) : null
+
+  const [reflectInitialized, setReflectInitialized] = useState(false)
+
+  const realmIcon = realmMapData?.partnerIcons.default ?? null
+  const stakingRealmColor = realmMapData?.color ?? "#2C2C2C"
+  const cursorTextColor = realmMapData?.cursorText ?? "#FFF"
+
+  useEffect(() => {
+    const initReflect = async () => {
+      if (reflectInitialized) return
+
+      await reflectInstance.mutate.initClientState({
+        id: reflectInstance.clientID,
+        cursor: null,
+        userInfo: {
+          name,
+          realmIcon,
+          stakingRealmColor,
+          cursorTextColor,
+        },
+        isPresent: true,
+      })
+
+      setReflectInitialized(true)
+    }
+
+    const updateUserInfo = async () => {
+      await reflectInstance.mutate.setUserInfo({
+        name,
+        realmIcon,
+        stakingRealmColor,
+        cursorTextColor,
+      })
+    }
+
+    initReflect()
+    updateUserInfo()
+
+    const handleReflectCursor = async (e: MouseEvent) => {
+      await reflectInstance.mutate.setCursor({ x: e.clientX, y: e.clientY })
+    }
+
+    window.addEventListener("mousemove", handleReflectCursor)
```

Comment:
Let's create another useEffect to handle event listeners, without dependencies, just on initialize, it'll improve the readability of this code.